### ISSUE TITLE
User argo-rollouts cannot update deployments in API group apps in the namespace X

### ISSUE THREAD

**Issue by amorattab** [argoproj/argo-helm#2826]
### Describe the bug

When using `workloadRef` to refer an existing deployment, and set the `scaleDown: onsuccess`, the controller tries to scale down the existing deployment after the rollout is done successfully, but it does not have the proper permissions and it shows the following error:

> msg="deployments.apps \"test-app\" is forbidden: User \"system:serviceaccount:argo-rollouts:argo-rollouts\" cannot update resource \"deployments\" in API group \"apps\" in the namespace \"test\"" error="<nil>"

I checked the clusterrole bounded to the argo-rollouts service account used by the controller pods and here is the list of permissions on deployments:

```
% kubectl get clusterrole argo-rollouts -o yaml
...
- apiGroups:
  - ""
  - apps
  resources:
  - deployments
  - podtemplates
  verbs:
  - get
  - list
  - watch
...
```
I edited the clusterrole added the `- update` and that fixed the problem.

### Related helm chart

argo-rollouts

### Helm chart version

2.37.2

### To Reproduce

1. install the rollout with the default values.yaml
2. Create a test-deployment resource with at least 1 replica.
3. Create a rollout manifest with reference to the existing test-deployment:
```
apiVersion: argoproj.io/v1alpha1                                                                                               
kind: Rollout
...
  workloadRef:
    apiVersion: apps/v1
    kind: Deployment
    name: test-deployment
    scaleDown: onsuccess
...
```

### Expected behavior

The number of pods on test-deploment scale-downs to 0 after a successfule rollout.

### Additional context

The error I get from the argo-rollouts controller pods is:
`msg="deployments.apps \"sama-tasks\" is forbidden: User \"system:serviceaccount:argo-rollouts:argo-rollouts\" cannot update resource \"deployments\" in API group \"apps\" in the namespace \"samaworld-amorattab\"" error="<nil>"`

**Comment by yu-croco** [comment]
Hi @amorattab , thank you for opening an issue.
It's duplicated as https://github.com/argoproj/argo-helm/issues/2795 , so let us close this.

We follow [upstream](https://github.com/argoproj/argo-rollouts/blob/v1.7.1/manifests/role/argo-rollouts-clusterrole.yaml#L59-L69) and it's not yet fixed in v1.7.1. When upstream releases fixed version, we will follow soon.

**Comment by amorattab** [comment]
> Hi @amorattab , thank you for opening an issue. It's duplicated as #2795 , so let us close this.
> 
> We follow [upstream](https://github.com/argoproj/argo-rollouts/blob/v1.7.1/manifests/role/argo-rollouts-clusterrole.yaml#L59-L69) and it's not yet fixed in v1.7.1. When upstream releases fixed version, we will follow soon.

Thanks @yu-croco for your response! realized that addtional RBAC rules can be added to the cluster role through providerRBAC.additionalRules.
